plaidml/bridge/openvino/ops/detection_output.cpp

```diff
@@ -0,0 +1,464 @@
+    auto share_location = layer->get_attrs().share_location;
+    auto nms_threshold = layer->get_attrs().nms_threshold;
+    auto confidence_threshold = layer->get_attrs().confidence_threshold;
+    auto clip_after_nms = layer->get_attrs().clip_after_nms;
+    auto clip_before_nms = layer->get_attrs().clip_before_nms;
+    auto decrease_label_id = layer->get_attrs().decrease_label_id;
+    auto normalized = layer->get_attrs().normalized;
+    auto input_height = layer->get_attrs().input_height;
+    auto input_width = layer->get_attrs().input_width;
+    auto objectness_score = layer->get_attrs().objectness_score;
+
+    int prior_size = normalized ? 4 : 5;
+    int prior_offset = normalized ? 0 : 1;
+    int i_h = normalized ? 1 : input_height;
+    int i_w = normalized ? 1 : input_width;
+    auto batch = Location.compute_shape().sizes()[0];
+    auto num_priors = Priors.compute_shape().sizes()[2] / prior_size;
+    auto priors_shape_variance = Priors.compute_shape().sizes()[1];
+
+    std::vector<int64_t> location_shape = {batch, num_priors, 4};
+    std::vector<int64_t> confidence_shape = {batch, num_priors, num_classes};
+    std::vector<int64_t> priors_shape = {batch, priors_shape_variance, num_priors, prior_size};
+
+    edsl::Tensor location = edsl::reshape(Location, location_shape);
+    edsl::Tensor confidence = edsl::reshape(Confidence, confidence_shape);
+    edsl::Tensor priors = edsl::reshape(Priors, priors_shape);
+
+    edsl::Tensor variances;
+    if (variance_encoded_in_target) {
+      // Fill variances with 1s.
+      priors =
+          op::slice(priors).add_dim(0, batch).add_dim(0, 1).add_dim(0, num_priors).add_dim(prior_offset, prior_size);
+      auto one = cast(edsl::Tensor{1}, priors.dtype());
+      auto var_shape = priors.compute_shape().sizes();
+      std::vector<int> target_shape(begin(var_shape), end(var_shape));
+      std::vector<int> target_axes = {};
+      variances = op::broadcast(one, target_shape, target_axes);
+    } else {
+      variances = op::slice(priors).add_dim(0, batch).add_dim(1, 2).add_dim(0, num_priors).add_dim(0, prior_size);
+      // Reshape the variances tensor from {batch, 1, num_priors, 5} to {batch, 1, num_priors, 4};
+      variances = edsl::reshape(variances, {batch, 1, num_priors * prior_size});
+      // Truncate the variances tensor.
+      variances = op::slice(variances).add_dim(0, batch).add_dim(0, 1).add_dim(0, num_priors * 4);
+      variances = edsl::reshape(variances, {batch, 1, num_priors, 4});
+      priors =
+          op::slice(priors).add_dim(0, batch).add_dim(0, 1).add_dim(0, num_priors).add_dim(prior_offset, prior_size);
+    }
+
+    variances = op::squeeze(variances, {1});
+    priors = op::squeeze(priors, {1});
+
+    edsl::Tensor decoded_bboxes;
+    if (with_add_pred) {
+      // Update confidence if there are 5 inputs.
+      std::vector<edsl::Tensor> arm_conf_vec;
+      for (int i = 1; i < num_priors * 2; i += 2) {
```

Comment:
Can we use something like the following instead of for loop?
```
Tensor IX=edsl::index({edsl::TensorDim(num_priors)}, 0);
IX=IX*2+1;
edsl::Tensor arm_conf = edsl::gather(ArmConfidence), IX).axis(1);
```

Reply:
Done.